drivers/flash/spi_nor.c

```diff
@@ -915,27 +928,20 @@ static int spi_nor_erase(const struct device *dev, off_t addr, size_t size)
 				}
 			}
 			if (bet != NULL) {
-				spi_nor_cmd_addr_write(dev, bet->cmd, addr, NULL, 0);
+				ret = spi_nor_cmd_addr_write(dev, bet->cmd, addr, NULL, 0);
 				addr += BIT(bet->exp);
 				size -= BIT(bet->exp);
 			} else {
 				LOG_DBG("Can't erase %zu at 0x%lx",
 					size, (long)addr);
 				ret = -EINVAL;
 			}
+			if (ret != 0) {
+				break;
+			}
 		}
 
-#ifdef __XCC__
```

Comment:
This needs separate commit referencing the issue https://github.com/zephyrproject-rtos/zephyr/issues/43745

Reply:
I'm not sure what exactly was going on with the previous code on that platform? Possibly it just didn't like the return value being discarded there? If so, it should presumably no longer be an issue. Not sure if the CI infrastructure tests that build or not..